### PR TITLE
added script to download/clone 3rd party libs.

### PR DESCRIPTION
**Pull request by janiversen**
Using 1 script instead of having the downloads in all
build.sh ensures consistency and lowers maintenance.

Signed-off-by: Jan Iversen <jani@apache.org>

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [x ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

get-dep-lib.sh supposed to be integrated in the different build.sh scripts, so that
we only have 1 file to maintain in regard to 3rd libraries.

Windows seems to have its own life, so currently not part of the script.

test can be done like

mkdir test; cd test
../subsurface/scripts/get-dep-lib android|desktop|ios .

If/when this PR is accepted, build.sh will be changed.

### CODE REVIEW

**Comment by janiversen** [comment]
Wonder, what would be the right prefix in the commit message for something general ?

**Comment by dirkhh** [comment]
> Wonder, what would be the right prefix in the commit message for something general ?

We haven't been consistent in that. Sometimes it's things like "build system:" - but nothing is fine, too. Once we have established a pattern, it's nicer to stick with it (as it makes parsing the git log easier - changes that affect only one platform have such an established pattern, or changes that are just about Travis/CI).

File: scripts/get-dep-lib.sh [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+if [[ $BUILD = *"googlemaps"* ]]; then
+	if [ ! -d googlemaps ] ; then
+		git clone https://github.com/Subsurface-divelog/googlemaps.git
+	fi
+    pushd googlemaps
+	git fetch origin
+    git checkout master
+    git pull --rebase
+	popd
```
Comment (dirkhh) [inline scripts/get-dep-lib.sh]: space vs tab?

File: scripts/get-dep-lib.sh [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+PLATFORM=$1
+INSTDIR=$2
+if [ ! -d ${INSTDIR} ] ; then
+	echo "creating dir"
+	mkdir -p ${INSTDIR}
+fi
+
+case ${PLATFORM} in
+	desktop)
+		BUILD="libzip libgit2 googlemaps hidapi libcurl libusb openssl libssh2"
```
Comment (dirkhh) [inline scripts/get-dep-lib.sh]: I think that's a bit misleading - the Linux builds and Windows builds don't download these...
This PR doesn't include the intended USE of this script (which is fine), so I'm not sure how you are using this (maybe there's another PR for that already?), but "desktop" seems like a misleading choice.

**Review by dirkhh — APPROVED** [review]
I will refrain from starting new "bring me a rock" exercises and simply accept this. I have some vague ideas how this could evolve, but as it is it's good.

File: scripts/get-dep-lib.sh (hunk 1 above)
Comment (janiversen) [reply scripts/get-dep-lib.sh]: Ups. will correct

**Comment by janiversen** [comment]
I am open for modifications, but lso think this script is somethung that will evolve as time goes on

**Comment by dirkhh** [comment]
> I am open for modifications, but lso think this script is somethung that will evolve as time goes on

Based on recent experience I'd rather not do that in a PR. I'd appreciate a version with clean whitespace and then a PR that uses this script. If I have time later today (while doing my day job and figuring out what's broken with the EON Steel on Mac) I may tweak this a little :-)

**Comment by janiversen** [comment]
I updated the commit message to clearly state how it is intended to be used.

Feel free to tweak it as you like, you know better than me what is "correct". To me the only point was to have a single place where we decide what to download.

Tweak it directly in my commit or do as you did last, close this PR and open a new one.

Disclaimer: I have some ideas of how different e.g. Mac/iOS could share the downloads, but that is future.

I will prepare a PR, with the changes to the 3 build.sh (each with its own commit) later this weekend. I am right now working on a idea I have for tables/phones).

Question: is FULLSCREEN_SUPPORT and !NO_USERMANUAL ever used, seems to me as being dead code ?

**Comment by dirkhh** [comment]
FULLSCREEN_SUPPORT was a conditional compile added by @neolit123 in 3f7900bf4e382cc57133e307efeaa3856c90c505 as a fix / reminder to something that's broken.
The user manual viewer can be disabled - that's what this option is for.

**Comment by dirkhh** [comment]
Manually merged for one remaining whitespace fix plus tiny edits in commit message

**Pull request closed without merging**